include/sparrow/utils/repeat_container.hpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iterator>
+#include <ranges>
+
+namespace sparrow
+{
+    template <class T>
+    class repeat_view : public std::ranges::view_interface<repeat_view<T>>
+    {
+    public:
+
+        class iterator
```

Comment:
I just realized that this should inherit from [iterator_base](https://github.com/man-group/sparrow/blob/main/include/sparrow/utils/iterator.hpp#L378) to reduce the code size. This can be done in a dedicated PR though.